test/functional/apps/dashboard/_dashboard_state.js

```diff
@@ -219,5 +128,108 @@ export default function ({ getService, getPageObjects }) {
 
       expect(changedTileMapData.length).to.not.equal(tileMapData.length);
     });
+
+    describe('Directly modifying url updates dashboard state', () => {
+      it('for query parameter', async function () {
+        await PageObjects.dashboard.gotoDashboardLandingPage();
+        await PageObjects.dashboard.clickNewDashboard();
+
+        const currentQuery = await PageObjects.dashboard.getQuery();
+        expect(currentQuery).to.equal('');
+        const currentUrl = await remote.getCurrentUrl();
+        const newUrl = currentUrl.replace('query:%27%27', 'query:%27hi%27');
+        // Don't add the timestamp to the url or it will cause a hard refresh and we want to test a
+        // soft refresh.
+        await remote.get(newUrl.toString(), false);
+        const newQuery = await PageObjects.dashboard.getQuery();
+        expect(newQuery).to.equal('hi');
+      });
+
+      it('for panel size parameters', async function () {
+        await PageObjects.dashboard.addVisualization(PIE_CHART_VIS_NAME);
+        const currentUrl = await remote.getCurrentUrl();
+        const currentPanelDimensions = await PageObjects.dashboard.getPanelDimensions();
+        const newUrl = currentUrl.replace(`w:${DEFAULT_PANEL_WIDTH}`, `w:${DEFAULT_PANEL_WIDTH * 2}`);
+        await remote.get(newUrl.toString(), false);
+        await retry.try(async () => {
+          const newPanelDimensions = await PageObjects.dashboard.getPanelDimensions();
+          if (newPanelDimensions.length < 0) {
+            throw new Error('No panel dimensions...');
+          }
+          // Some margin of error is allowed (I've noticed it being off by one pixel which probably something to do
+          // with an odd width and dividing by two), but due to https://github.com/elastic/kibana/issues/14542 I'm
+          // adding more margin of error than should be necessary.  That issue looks legit, but because I can't
+          // repro locally, I don't have a quick solution aside from increasing this margin error, for getting the
+          // build to pass consistently again.
+          const marginOfError = 20;
+          expect(newPanelDimensions[0].width).to.be.lessThan(currentPanelDimensions[0].width * 2 + marginOfError);
+          expect(newPanelDimensions[0].width).to.be.greaterThan(currentPanelDimensions[0].width * 2 - marginOfError);
+        });
+      });
+
+      it('when removing a panel', async function () {
```

Comment:
This big chunk of code I just moved around to reflect the actual order it's run in (unnested it's before describe blocks).  but I did add this test bc of a bug I found in the PR which wasnt retruning true for `_areStoreAndAppStatePanelsEqual` when one had less panels than the other.